lib/Component.lua

```diff
@@ -106,12 +106,9 @@ end
 	current state object.
 ]]
 function Component:setState(partialState)
-	-- State cannot be set in any of the following places:
-	-- * During the component's init function
-	-- * During the component's render function
-	-- * After the component has been unmounted (or is in the process of unmounting, e.g. willUnmount)
+	-- State cannot be set in any lifecycle hooks.
 	if not self._canSetState then
-		error("setState cannot be used currently: are you calling setState from an init, render, or willUnmount function?", 0)
+		error("setState cannot be used currently: are you calling setState within a lifecycle hook?", 0)
```

Comment:
I like mentioning `render` and `init` explicitly here since they aren't lifecycle hooks.

Reply:
I've changed the error message to be more descriptive:

```
setState cannot be used currently, are you calling setState from any of:
* the willUpdate or willUnmount lifecycle hooks
* the init function
* the render function
* the shouldUpdate function
```